src/foam/nanos/approval/ApprovalRequest.js

```diff
@@ -0,0 +1,440 @@
+      visibility: function(requiredPoints) {
+        return requiredPoints ?
+          foam.u2.DisplayMode.RO :
+          foam.u2.DisplayMode.HIDDEN;
+      }
+    },
+    {
+      class: 'Int',
+      name: 'requiredRejectedPoints',
+      value: 1,
+      gridColumns: 4,
+      section: 'basicInformation',
+      visibility: function(requiredRejectedPoints) {
+        return requiredRejectedPoints ?
+          foam.u2.DisplayMode.RO :
+          foam.u2.DisplayMode.HIDDEN;
+      }
+    },
+    {
+      class: 'Reference',
+      of: 'foam.nanos.auth.Group',
+      name: 'group',
+      documentation: `When set, each user in the group will receive a request for approval.
+      If "approver" property is set, "group" property is ignored.`,
+      section: 'supportDetails',
+      visibility: function(group) {
+        return group ?
+          foam.u2.DisplayMode.RO :
+          foam.u2.DisplayMode.HIDDEN;
+      }
+    },
+    {
+      class: 'Enum',
+      of: 'foam.nanos.approval.ApprovalStatus',
+      name: 'status',
+      value: 'REQUESTED',
+      section: 'requestDetails',
+      javaFactory: 'return foam.nanos.approval.ApprovalStatus.REQUESTED;',
+      visibility: function(status) {
+        return status ?
+          foam.u2.DisplayMode.RO :
+          foam.u2.DisplayMode.HIDDEN;
+      }
+    },
+    {
+      class: 'String',
+      name: 'memo',
+      view: { class: 'foam.u2.tag.TextArea', rows: 5, cols: 80 },
+      documentation: 'Meant to be used for explanation on why request was approved/rejected',
+      section: 'basicInformation',
+      visibility: function(memo) {
+        return memo ?
+          foam.u2.DisplayMode.RO :
+          foam.u2.DisplayMode.HIDDEN;
+      }
+    },
+    {
+      class: 'String',
+      name: 'description',
+      documentation: `Approval request description.`,
+      tableWidth: 200,
+      section: 'basicInformation',
+      visibility: function(description) {
+        return description ?
+          foam.u2.DisplayMode.RO :
+          foam.u2.DisplayMode.HIDDEN;
+      }
+   },
+    {
+      class: 'String',
+      name: 'token',
+      documentation: 'token in email for ‘click to approve’.',
+      section: 'basicInformation',
+      visibility: 'HIDDEN'
+    },
+    {
+      class: 'DateTime',
+      name: 'created',
+      section: 'supportDetails',
+      gridColumns: 6,
+      visibility: function(created) {
+        return created ?
+          foam.u2.DisplayMode.RO :
+          foam.u2.DisplayMode.HIDDEN;
+      }
+    },
+    {
+      class: 'DateTime',
+      name: 'lastModified',
+      gridColumns: 6,
+      section: 'supportDetails',
+      visibility: function(lastModified) {
+        return lastModified ?
+          foam.u2.DisplayMode.RO :
+          foam.u2.DisplayMode.HIDDEN;
+      }
+    },
+    {
+      class: 'Reference',
+      of: 'foam.nanos.auth.User',
+      name: 'createdBy',
+      section: 'supportDetails',
+      visibility: 'HIDDEN'
```

Comment:
Why are these HIDDEN? 
I've been creating an 'audit' section for these type of properties. Can be permissioned.

Reply:
Was last minute fix, just changed it to permissionRequired: true